src/features/unified-share-modal/EmailForm.js

```diff
@@ -59,28 +64,52 @@ type Props = {
 
 type State = {
     contactsFieldError: string,
+    justificationReasonsFieldError: string,
     message: string,
+    selectedJustificationReason: ?SelectOptionProp,
 };
 
 class EmailForm extends React.Component<Props, State> {
     static defaultProps = {
         messageProps: {},
         contactsFieldDisabledTooltip: null,
+        justificationReasons: [],
+        restrictedExternalEmails: [],
+        shouldRequireExternalContactJustification: false,
     };
 
     constructor(props: Props) {
         super(props);
 
         this.state = {
             contactsFieldError: '',
+            justificationReasonsFieldError: '',
             message: '',
+            selectedJustificationReason: null,
         };
     }
 
     contactsFieldRef: {
         current: null | PillSelectorDropdown,
     } = React.createRef();
 
+    componentDidUpdate(prevProps: Props, prevState: State) {
+        const { contactsFieldError, justificationReasonsFieldError } = this.state;
+        const {
+            contactsFieldError: prevContactsFieldError,
+            justificationReasonsFieldError: prevJustificationReasonsFieldError,
+        } = prevState;
+
+        // Only display one type of error at a time and give preference
```

Comment:
I also tried another approach in which there was a single `formError` which had a type that indicated which field it belonged to. Unfortunately with that approach the general validation that happens on submit would result in cases in which all errors would end up being hidden (depending on the validator order). It seemed to me that working around that resulted in a flow that was harder reason about, so settled for this approach instead.